CIP-0030/README.md

```diff
@@ -227,12 +227,23 @@ Errors: `APIError`
 
 Returns the reward addresses owned by the wallet. This can return multiple addresses e.g. CIP-0018.
 
-### api.signTx(tx: cbor\<transaction>, partialSign: bool = false): Promise\<cbor\<transaction_witness_set>>
+### 👎 api.signTx(tx: cbor\<transaction>, partialSign: bool = false): Promise\<cbor\<transaction_witness_set>>
 
 Errors: `APIError`, `TxSignError`
 
 Requests that a user sign the unsigned portions of the supplied transaction. The wallet should ask the user for permission, and if given, try to sign the supplied body and return a signed transaction. If `partialSign` is true, the wallet only tries to sign what it can. If `partialSign` is false and the wallet could not sign the entire transaction, `TxSignError` shall be returned with the `ProofGeneration` code. Likewise if the user declined in either case it shall return the `UserDeclined` code. Only the portions of the witness set that were signed as a result of this call are returned to encourage dApps to verify the contents returned by this endpoint while building the final transaction.
 
+
+### ⚗️ api.signTx(request: { tx: cbor\<transaction>, partialSign?: bool = false, inputHints?: cbor\<transaction>[] = [] }): Promise\<cbor\<transaction_witness_set>>
+
+Errors: `APIError`, `TxSignError`
+
+Requests that a user sign the unsigned portions of the supplied transaction. The wallet should ask the user for permission, and if given, try to sign the supplied body and return a signed transaction.
+
+- `partialSign`: if true, the wallet only tries to sign what it can. If false and the wallet could not sign the entire transaction, `TxSignError` shall be returned with the `ProofGeneration` code. Likewise if the user declined in either case it shall return the `UserDeclined` code. Only the portions of the witness set that were signed as a result of this call are returned to encourage dApps to verify the contents returned by this endpoint while building the final transaction.
+
+- `inputHints`: to allow dApps to chain transactions together, they may need to provide to the wallet hints about some of the inputs of the transaction that are not on-chain yet. In order to do this in a trustless way, we have to provide the full transaction body (so that the wallet can hash the tx body and compare it to the hash in the inputs). Providing the full transaction instead of just the transaction body also allows access to any scripts or data in the witness that may be useful for the wallet. Note these hints could be use by the wallet to resolve regular transaction inputs, collateral or any other feature of Cardano in the future
```

Comment:
The main part I'm not entirely sure about is whether or not these input hints should be the full transaction or just the tx body. For chaining transactions, you just need the tx body. For things like data hints of script hints, we could make these separate fields instead of putting this all as part of the same field.  Open to thoughts on this.

Reply:
I have a question, why can't we just sign continuing txs by calling `signTx` multiple times? Does the wallet implementation check for existing input and throw error if inputs don't exist whenever we call `signTx`? If this is the case, then isn't the solution is to change the wallet implementation to not check for existing inputs?
On a side note, to support chaining tx, I think we would also need support on the infrastructure side (i.e. Blockfrost) to submit multiple txs in a single API call to the same node

Reply:
@longngn if the wallet doesn't have access to the input, it can't know which input should be signed with which key

Reply:
I asked around and it seems nobody needs the full tx so just the tx body should be enough for everyone

Reply:
To present the user will all information about a transaction, including metadata, I would like to see transactions instead of transaction bodies as hints.

Reply:
Apart from Allowing to review all non on-chain transactions at once that let to this input, no.